src/buttons/buttonPreview.js

```diff
@@ -15,9 +17,9 @@ export default class ButtonPreview {
    * Add the preview button
    */
   addButton() {
-    const emailForm = ButtonPreviewCommand.getEmailForm();
-    const url = emailForm[0].action;
-    const regexp = /emails\/new/g;
+    const form = ButtonsService.getForm();
+    const url = form[0].action;
+    const regexp = /(emails|pages)\/new/g;
```

Comment:
This is not best practice. Like this, we can not change the URL. And we would have to way of finding out that this will break. Please always try to only rely on values that are available inside the class. Or if not possible rely on existing solutions. So we at least don't break more things :)

I think this can help: Mautic.isNewEntity('#page_sessionId, #emailform_sessionId');
it is in 4.builder.js